src/Symfony/Component/Routing/Generator/ConfigurableQueryEncTypeInterface.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Routing\Generator;
+
+/**
+ * ConfigurableQueryEncTypeInterface must be implemented by URL generators that
+ * can be configured to build query strings by replacing spaces with their
+ * percent-encoded value (%20) according to RFC3986 or by replacing them with a
+ * plus sign (+) according to RFC1738.
+ *
+ * @author Jonathan Cox <jonathan.cox@fostermade.co>
+ */
+interface ConfigurableQueryEncTypeInterface
```

Comment:
For clarity I recommend to name it `ConfigurableQueryEncodingTypeInterface` with the method `setQueryEncodingType`, I'm not sure if `EncType` is a standard name here though.